java/server/test/org/openqa/selenium/grid/router/ProxyCdpTest.java

```diff
@@ -126,6 +128,41 @@ public void onText(CharSequence data) {
     socket.close();
   }
 
+  @Test
+  public void shouldBeAbleToSendMessagesOverSecureWebSocket()
+      throws URISyntaxException, InterruptedException {
+    HttpClient.Factory clientFactory = HttpClient.Factory.createDefault();
+
+    Config secureConfig = new MapConfig(ImmutableMap.of("node", ImmutableMap.of("--self-signed-https","","https-private-key", "", "https-certificate", "")));
```

Comment:
This isn't how to set the self-signed cert flag. You need:

```java
ImmutableMap.of(
  "server", ImmutableMap.of(
    "https-self-signed", true));
```

TL;DR: the flags are different from the Config settings. You can see the mapping in `BaseServerFlags`